### ISSUE TITLE
One-hot encoded categorical values in model

### ISSUE THREAD

**Issue by Ullar-Kask** [tensorflow/recommenders#630]
Hi,

In my model I am trying to use some categorical inputs transformed into embeddings (Embedding) and some transformed into one-hot encoded form (e.g. person's gender). I am unable to concat the values, `tf.keras.layers.Embedding()` and `tf.keras.layers.IntegerLookup(output_mode='one_hot')` seem to produce outputs with incompatible dimensions. `tf.concat([embedding, one_hot], axis=1)` yields 
```
ConcatOp : Ranks of all input tensors should match: shape[0] = [1,16] vs. shape[2] = [13] [Op:ConcatV2] name: concat
```
What I am missing?
The interesting fact is that this is happening only during inference when I query the index. No errors during training.

**Comment by patrickorlando** [comment]
I encountered this problem too @Ullar-Kask.
It appears that the batch dimension is omitted when a single example is passed in.

```python
one_hot_encoding = tf.keras.layers.CategoryEncoding(num_tokens=4, output_mode="one_hot")
inputs = tf.range(4)
outputs = one_hot_encoding(inputs)
```
results in 
```
<tf.Tensor: shape=(4, 4), dtype=float32, numpy=
array([[1., 0., 0., 0.],
       [0., 1., 0., 0.],
       [0., 0., 1., 0.],
       [0., 0., 0., 1.]], dtype=float32)>
```
but
```python
inputs = tf.constant([2])
outputs = one_hot_encoding(inputs)
```
gives
```
<tf.Tensor: shape=(4,), dtype=float32, numpy=array([0., 0., 1., 0.], dtype=float32)>
```
which only has 1 dimension.

My solution has been to implement a custom layer for one hot encoding, but I'm interested to hear if anyone else has a better solution.

```python
class OneHotEncoding(tf.keras.layers.Layer):
    def __init__(self, vocabulary_size, mask_zero=False, **kwargs):
        super().__init__(**kwargs)
        self._vocabulary_size = vocabulary_size
        self._mask_zero = mask_zero

    def call(self, inputs):
        return tf.one_hot(inputs, depth=self._vocabulary_size)

    def compute_mask(self, inputs, mask=None):
        if self._mask_zero:
            return tf.logical_not(tf.equal(inputs, tf.constant(0, dtype=inputs.dtype)))

    def get_config(self):
        config = {
            "vocabulary_size": self._vocabulary_size,
            "mask_zero": self._mask_zero,
        }
        return config

one_hot_encoding = OneHotEncoding(vocabulary_size=4)
inputs = tf.constant([2])
outputs = one_hot_encoding(inputs)
```
which gives
```
<tf.Tensor: shape=(1, 4), dtype=float32, numpy=array([[0., 0., 1., 0.]], dtype=float32)>
```

**Comment by Ullar-Kask** [comment]
Thanka a lot @patrickorlando !
